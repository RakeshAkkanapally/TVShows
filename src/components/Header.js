import { AppBar, Grid, makeStyles, Toolbar, Typography } from '@material-ui/core'
import Paper from '@material-ui/core/Paper';
import MovieIcon from '@material-ui/icons/Movie';
import { NavLink } from 'react-router-dom';


const useStyles = makeStyles(theme => ({
    root: {
        backgroundColor: '#000',
        color: '#000'
    },
    paper: {
        backgroundColor: 'inherit',
        textAlign: 'center',
        color: '#fff',
    },
    activeLink: {
        fontWeight: 'bold',
        color: 'blue',
    },
    logo: {
        textDecoration: 'none',
        backgroundColor: 'inherit',
        textAlign: 'center',
        color: '#fff',
    },
    hover: {
        '&:hover': {
            color: "#f00",
        }
    }

}));




function Header() {
    const classes = useStyles();

    return (
        <div>
            <AppBar position="static"
                className={classes.root}>
                <Grid container
                    direction="row"
                    alignItems="center">
                    <Grid item xs={9} >
                        <NavLink exact to="/" className={classes.logo}>
                                <Toolbar>
                                    <MovieIcon fontSize="large" />
                                    <Typography variant="h6">
                                        <strong>TV SHOWS</strong>
                                    </Typography>
                                </Toolbar>
                        </NavLink>
                    </Grid>
                    <Grid item xs >
                        <Paper className={classes.paper}>
                            <NavLink exact to="/home" activeClassName={classes.activeLink}>
                                <Typography className={classes.hover}>
                                    Home
                                </Typography>
                            </NavLink>
                        </Paper>
                    </Grid>
                    <Grid item xs >
                        <Paper className={classes.paper}>
                            <NavLink exact to="/" activeClassName={classes.activeLink}>
                                <Typography className={classes.hover}>
                                    Dashboard
                                </Typography>
                            </NavLink>
                        </Paper>
                    </Grid>
                </Grid>
            </AppBar>
        </div>)
}

export default Header;