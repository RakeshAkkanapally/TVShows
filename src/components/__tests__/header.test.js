import React from "react";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Header from "../Header";
import ShowContext from "../../reducers/showContext";



it("renders correctly", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <ShowContext.Provider value={ShowContext}>
          <Header />
        </ShowContext.Provider>
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
