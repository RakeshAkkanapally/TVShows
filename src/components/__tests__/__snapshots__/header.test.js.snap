// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-root-1 MuiPaper-elevation4"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9"
      >
        <a
          aria-current="page"
          className="makeStyles-logo-4 active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          <div
            className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M18 4l2 4h-3l-2-4h-2l2 4h-3l-2-4H8l2 4H7L5 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V4h-4z"
              />
            </svg>
            <h6
              className="MuiTypography-root MuiTypography-h6"
            >
              <strong>
                TV SHOWS
              </strong>
            </h6>
          </div>
        </a>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
      >
        <div
          className="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <a
            aria-current={null}
            href="/home"
            onClick={[Function]}
          >
            <p
              className="MuiTypography-root makeStyles-hover-5 MuiTypography-body1"
            >
              Home
            </p>
          </a>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
      >
        <div
          className="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <a
            aria-current="page"
            className="makeStyles-activeLink-3"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            <p
              className="MuiTypography-root makeStyles-hover-5 MuiTypography-body1"
            >
              Dashboard
            </p>
          </a>
        </div>
      </div>
    </div>
  </header>
</div>
`;
